thetis/callback.py

```diff
@@ -707,7 +750,12 @@ def __call__(self):
             try:
                 field = self.solver_obj.fields[fieldname]
                 if self.eval_func is None:
-                    val = field.at(self.xyz, tolerance=self.tolerance)
+                    P0DG, P0DG_input_ordering = self.function_spaces[fieldname]
+                    f_at_points = Function(P0DG)
+                    f_at_input_points = Function(P0DG_input_ordering)
+                    f_at_points.interpolate(field)
+                    f_at_input_points.interpolate(f_at_points)
```

Comment:
Do you know if this vom evaluation works for 2D mesh on the sphere (in which case the point coordinates are 3d x,y,z)? The whole `eval_func` option was added to enable `at()`-like evaluation in this case (`eval_func` was a modified version of Firedrake's `at`)

Reply:
Unsure as I only usually do boring plain 2D meshes! I mainly fixed because of the parallelisation issues for big simulations. Happy to look into it in the future or perhaps someone in the Firedrake group might have answer for this already.

Reply:
OK, we can address it in another PR.